src/Nuxeo/Client/Api/NuxeoClient.php

```diff
@@ -68,16 +68,39 @@ class NuxeoClient {
    * @param string $password
    * @throws NuxeoClientException
    */
-  public function __construct($url = 'http://localhost:8080/nuxeo', $username = 'Administrator', $password = 'Administrator') {
+  public function __construct($url = 'http://localhost:8080/nuxeo', $username = 'Administrator', $password = 'Administrator', $authType = 'BASIC', $portalKey = "") {
     try {
       $this->baseUrl = Url::factory($url);
-      $this->httpClient = new Client($url, array(
-        Client::REQUEST_OPTIONS => array(
-          'headers' => array(
-            'Content-Type' => 'application/json+nxrequest'
-          )
-        )
-      ));
+      switch($authType) {
+         default:
+         case 'BASIC':
+            $this->httpClient = new Client($url, array(
+              Client::REQUEST_OPTIONS => array(
+                'headers' => array(
+                  'Content-Type' => 'application/json+nxrequest'
+                )
+              )
+            ));
+           break;
+         case 'PORTAL':
+           $timestamp = time() * 1000;
+           $random = rand(0, $timestamp);
```

Comment:
As we use that random in the token generation, it would be safer to use random_int from PHP7.
Reference: http://php.net/manual/en/function.rand.php
Polyfill available at https://github.com/paragonie/random_compat